micromanager_gui/_core_widgets/_stage_widget.py

```diff
@@ -0,0 +1,225 @@
+            p = ", ".join(str(round(x, 2)) for x in pos)
+        elif self._dtype is DeviceType.AutoFocus:
+            p = ""
+        else:
+            p = round(self._mmc.getPosition(self._device), 2)
+        self._readout.setText(f"{self._device}:  {p}")
+
+    def _update_ttips(self):
+        coords = chain(zip(repeat(3), range(7)), zip(range(7), repeat(3)))
+
+        Y = {DeviceType.XYStage: "Y", DeviceType.AutoFocus: "Offset"}.get(
+            self._dtype, "Z"
+        )
+
+        btn_layout: QGridLayout = self._btns.layout()
+        for r, c in coords:
+            if item := btn_layout.itemAtPosition(r, c):
+                if (r, c) == (3, 3):
+                    continue
+                if btn := item.widget():
+                    xmag, ymag = self.BTNS[f"{PREFIX}.{btn.text()}"][-2:]
+                    if xmag:
+                        btn.setToolTip(f"move X by {self._scale(xmag)} µm")
+                    elif ymag:
+                        btn.setToolTip(f"move {Y} by {self._scale(ymag)} µm")
+
+    def _set_xy_visible(self):
+        if self._dtype is not DeviceType.XYStage:
+            btn_layout: QGridLayout = self._btns.layout()
+            for c in (0, 1, 2, 4, 5, 6):
+                if item := btn_layout.itemAtPosition(3, c):
+                    item.widget().hide()
+
+    def _set_visible_levels(self, levels: int):
+        """Hide upper-level stage buttons as desired. Levels must be between 1-3."""
+        assert 1 <= levels <= 3, "levels must be between 1-3"
+        self._levels = levels
+        btn_layout: QGridLayout = self._btns.layout()
+        for btn in self._btns.findChildren(QPushButton):
+            btn.show()
+        if levels < 3:
+            # hide row/col 0, 6
+            for r, c in product(range(7), (0, 6)):
+                if item := btn_layout.itemAtPosition(r, c):
+                    item.widget().hide()
+                if item := btn_layout.itemAtPosition(c, r):
+                    item.widget().hide()
+        if levels < 2:
+            # hide row/col 1, 5
+            for r, c in product(range(1, 6), (1, 5)):
+                if item := btn_layout.itemAtPosition(r, c):
+                    item.widget().hide()
+                if item := btn_layout.itemAtPosition(c, r):
+                    item.widget().hide()
+
+    def _on_click(self):
+        btn: QPushButton = self.sender()
+        xmag, ymag = self.BTNS[f"{PREFIX}.{btn.text()}"][-2:]
+        self._move_stage(self._scale(xmag), self._scale(ymag))
+
+    def _move_stage(self, x, y):
+        if self._dtype is DeviceType.XYStage:
+            self._mmc.setRelativeXYPosition(self._device, x, y)
+        elif self._dtype is DeviceType.AutoFocus:
+            self._mmc.setAutoFocusOffset(y)
+        else:
+            self._mmc.setRelativePosition(self._device, y)
+        self._mmc.waitForDevice(self._device)
+        if self._snap_on_click:
+            self._mmc.snap()
```

Comment:
Either there needs to be a change in core to have this stop live mode - or there there needs to be acheck here of if a sequence is running. This can result in this error:


```
untimeError                              Traceback (most recent call last)
File ~/Documents/oss/micro/napari-micromanager/micromanager_gui/_core_widgets/_stage_widget.py:207, in StageWidget._on_click(self=<micromanager_gui._core_widgets._stage_widget.StageWidget object>)
    205 btn: QPushButton = self.sender()
    206 xmag, ymag = self.BTNS[f"{PREFIX}.{btn.text()}"][-2:]
--> 207 self._move_stage(self._scale(xmag), self._scale(ymag))
        xmag = 2
        ymag = 0
        self = <micromanager_gui._core_widgets._stage_widget.StageWidget object at 0x7f5d33328ee0>

File ~/Documents/oss/micro/napari-micromanager/micromanager_gui/_core_widgets/_stage_widget.py:218, in StageWidget._move_stage(self=<micromanager_gui._core_widgets._stage_widget.StageWidget object>, x=20, y=0)
    216 self._mmc.waitForDevice(self._device)
    217 if self._snap_on_click:
--> 218     self._mmc.snap()
        self = <micromanager_gui._core_widgets._stage_widget.StageWidget object at 0x7f5d33328ee0>
        self._mmc = <CMMCorePlus at 0x7f5d3326cd10>

File ~/mambaforge/envs/pymm/lib/python3.10/site-packages/pymmcore_plus/core/_mmcore_plus.py:648, in CMMCorePlus.snap(self=<CMMCorePlus>, fix=True, *args=())
    629 def snap(self, *args, fix=True) -> np.ndarray:
    630     """
    631     snap and return an image.
    632 
   (...)
    646     img : np.ndarray
    647     """
--> 648     self.snapImage()
        self = <CMMCorePlus at 0x7f5d3326cd10>
    649     img = self.getImage()
    650     self.events.imageSnapped.emit(img)

File ~/mambaforge/envs/pymm/lib/python3.10/site-packages/wrapt/decorators.py:469, in synchronized.<locals>._synchronized(wrapped=<bound method CMMCorePlus.snapImage of <CMMCorePlus>>, instance=<CMMCorePlus>, args=(), kwargs={})
    463 @decorator
    464 def _synchronized(wrapped, instance, args, kwargs):
    465     # Execute the wrapped function while the original supplied
    466     # lock is held.
    468     with lock:
--> 469         return wrapped(*args, **kwargs)
        wrapped = <bound method CMMCorePlus.snapImage of <CMMCorePlus at 0x7f5d3326cd10>>
        args = ()
        kwargs = {}

File ~/mambaforge/envs/pymm/lib/python3.10/site-packages/pymmcore_plus/core/_mmcore_plus.py:553, in CMMCorePlus.snapImage(self=<CMMCorePlus>)
    551 @synchronized(lock)
    552 def snapImage(self) -> None:
--> 553     return super().snapImage()

RuntimeError: This operation can not be executed while sequence acquisition is running.
```

to reproduce that error:
1. launch
2. start live mode
3. trying moving stage

Reply:
I love this "snap_after_stage_moves" function but I think we can remove it for now and see how to add it back once the stage, the snap and the live button widgets are merged. Do you agree?